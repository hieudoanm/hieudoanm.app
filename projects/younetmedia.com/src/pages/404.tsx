import { ErrorTemplate } from '@younetmedia/templates/ErrorTemplate';
import { NextPage } from 'next';

export const NotFoundPage: NextPage = () => (
  <ErrorTemplate code={404} message="Page Not Found" />
);

export default NotFoundPage;
