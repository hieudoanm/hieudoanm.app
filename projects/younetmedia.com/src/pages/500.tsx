import { ErrorTemplate } from '@younetmedia/templates/ErrorTemplate';
import { NextPage } from 'next';

export const InternalServerErrorPage: NextPage = () => (
  <ErrorTemplate code={500} message="Internal Server Error" />
);

export default InternalServerErrorPage;
