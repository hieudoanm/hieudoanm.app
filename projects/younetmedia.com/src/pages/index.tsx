import { Navbar } from '@younetmedia/common/components/Navbar';
import { NextPage } from 'next';
import Link from 'next/link';

export const HomePage: NextPage = () => {
  return (
    <>
      <Navbar />
      <main className="container mx-auto p-8">
        <div className="grid grid-cols-2 gap-8">
          <div className="col-span-1">
            <Link href="/socialheat">
              <button
                className="btn bg-teal-500 text-white w-full"
                type="button">
                Social Heat
              </button>
            </Link>
          </div>
          <div className="col-span-1">
            <Link href="/socialtrend">
              <button
                className="btn bg-teal-500 text-white w-full"
                type="button">
                Social Trend
              </button>
            </Link>
          </div>
        </div>
      </main>
    </>
  );
};

export default HomePage;
