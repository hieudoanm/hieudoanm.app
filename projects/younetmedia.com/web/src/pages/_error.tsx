import { ErrorTemplate } from '@younetmedia/templates/ErrorTemplate';
import { NextPage } from 'next';

const InternalServerErrorPage: NextPage = () => (
  <ErrorTemplate code={500} message="Error" />
);

export default InternalServerErrorPage;
