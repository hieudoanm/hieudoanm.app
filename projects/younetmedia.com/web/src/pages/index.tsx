import { Button } from '@mui/material';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { Navbar } from '@younetmedia/components/Navbar';
import { NextPage } from 'next';
import Link from 'next/link';

export const HomePage: NextPage = () => {
  return (
    <LocalizationProvider dateAdapter={AdapterDayjs}>
      <Navbar />
      <main className="container mx-auto p-8">
        <div className="grid grid-cols-2 gap-8">
          <div className="col-span-1">
            <Link href="/socialheat">
              <Button fullWidth variant="outlined">
                Social Heat
              </Button>
            </Link>
          </div>
          <div className="col-span-1">
            <Link href="/socialtrend">
              <Button fullWidth variant="outlined">
                Social Trend
              </Button>
            </Link>
          </div>
        </div>
      </main>
    </LocalizationProvider>
  );
};

export default HomePage;
