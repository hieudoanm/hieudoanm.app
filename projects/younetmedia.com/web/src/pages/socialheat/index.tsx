import {
  Button,
  Paper,
  Stack,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  TextField,
  ToggleButton,
  ToggleButtonGroup
} from '@mui/material';
import { DesktopDatePicker, LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import {
  ERROR_MESSAGE_AUTHENTICATION,
  ERROR_MESSAGE_DATE_RANGE
} from '@younetmedia/common/constants';
import { logger } from '@younetmedia/common/libs/log';
import { Navbar } from '@younetmedia/components/Navbar';
import { queryResult } from '@younetmedia/services/younetmedia.service';
import { Result } from '@younetmedia/types';
import { Dayjs } from 'dayjs';
import { NextPage } from 'next';
import { useRouter } from 'next/router';
import React, { ChangeEvent, FormEvent, useState } from 'react';

export const SocialHeatPage: NextPage = () => {
  const router = useRouter();

  const [topicId, setTopicId] = useState<number>(0);
  const [fromDate, setFromDate] = useState<Dayjs>();
  const [toDate, setToDate] = useState<Dayjs>();
  const [input, setInput] = useState<'input' | 'table'>('input');
  const [queryString, setQueryString] = useState<string>('');
  const [queries, setQueries] = useState<string[]>([]);
  const [results, setResults] = useState<Result[]>([]);
  const [loading, setLoading] = useState<number>(0);

  const changeInput = (
    _event: React.MouseEvent<HTMLElement>,
    newInput: 'input' | 'table'
  ) => {
    setInput(newInput);
  };

  const changeFromDate = (newFromDate: Dayjs) => {
    setFromDate(newFromDate);
  };

  const changeToDate = (newToDate: Dayjs) => {
    setToDate(newToDate);
  };

  const changeTopicId = (
    event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    const newTopicId: number =
      Number.parseInt(event?.target.value || '0', 10) || 0;
    setTopicId(newTopicId);
  };

  const changeQueryString = (
    event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    const newQueryString = event?.target.value || '';
    setQueryString(newQueryString);
    setQueries(newQueryString.split('\n'));
  };

  const runQueries = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (fromDate === null || toDate === null) {
      alert(ERROR_MESSAGE_DATE_RANGE);
      return;
    }

    const accessToken: string = sessionStorage.getItem('accessToken') ?? '';
    if (!accessToken) {
      alert(ERROR_MESSAGE_AUTHENTICATION);
      router.push('/auth');
    }

    setLoading(0);
    setResults([]);
    const newResults = [];
    let index = 0;
    const total = queries.length - 1;
    for (const query of queries) {
      try {
        const result: Result = await queryResult(
          topicId,
          {
            fromDate: fromDate?.format('YYYY-MM-DD') ?? '',
            toDate: toDate?.format('YYYY-MM-DD') ?? ''
          },
          query
        );
        newResults.push(result);
      } catch (error) {
        logger.error(error);
        newResults.push({
          query,
          total_collectable_mentions: -1,
          total_mentions: -1
        } as Result);
      }
      setLoading(Number.parseFloat(((index / total) * 100).toFixed(2)));
      index++;
    }
    setResults(newResults);
  };

  const downloadCSV = () => {
    const headers: string[] = [
      'query',
      'total_collectable_mentions',
      'total_mentions'
    ];
    const headerRow: string = `${headers.join(',')}`;
    const dataRows: string = results
      .map((result: any) => {
        return headers
          .map((header: string) => {
            if (header === 'query') {
              const cell = result[`${header}`] || '';
              return `"${cell.replaceAll('"', "'").replaceAll('#', '')}"`;
            }
            return result[`${header}`];
          })
          .join(',');
      })
      .join('\n');
    const csvContent = `data:text/csv;charset=utf-8,${headerRow}\n${dataRows}`;
    const encodedUri: string = encodeURI(csvContent);
    const link: HTMLAnchorElement = document.createElement('a');
    link.setAttribute('href', encodedUri);
    link.setAttribute(
      'download',
      `${new Date().toISOString().split('T')[0]}.csv`
    );
    document.body.append(link); // Required for FF
    link.click(); // This will download the data file named "my_data.csv".
  };

  return (
    <LocalizationProvider dateAdapter={AdapterDayjs}>
      <Navbar />
      <main className="container mx-auto p-8">
        <section className="mb-8">
          <div className="rounded border p-8 shadow-2xl">
            <form onSubmit={runQueries}>
              <div className="grid grid-cols-12 items-center gap-8">
                <div className="col-span-12">
                  <div className="flex items-center justify-between">
                    <h2 className="text-lg uppercase">Queries</h2>
                    <Button type="submit" variant="outlined">
                      Run Queries
                    </Button>
                  </div>
                </div>
                <div className="col-span-12">
                  <TextField
                    type="number"
                    id="topicId"
                    label="Topic ID"
                    placeholder="Topic ID"
                    className="w-full"
                    value={topicId}
                    onChange={changeTopicId}
                    required
                  />
                </div>
                <div className="col-span-6">
                  <DesktopDatePicker
                    label="From Date"
                    format="DD/MM/YYYY"
                    value={fromDate}
                    onChange={(newDate: Dayjs | null) => {
                      if (!newDate) {
                        return;
                      }
                      changeFromDate(newDate);
                    }}
                    className="w-full"
                  />
                </div>
                <div className="col-span-6">
                  <DesktopDatePicker
                    label="To Date"
                    format="DD/MM/YYYY"
                    value={toDate}
                    onChange={(newDate: Dayjs | null) => {
                      if (!newDate) {
                        return;
                      }
                      changeToDate(newDate);
                    }}
                    className="w-full"
                  />
                </div>
                <div className="col-span-12">
                  <Stack spacing={2} alignItems="center">
                    <ToggleButtonGroup
                      size="large"
                      value={input}
                      onChange={changeInput}
                      exclusive={true}
                    >
                      <ToggleButton value="input" key="input">
                        Input
                      </ToggleButton>
                      <ToggleButton value="table" key="table">
                        Table
                      </ToggleButton>
                    </ToggleButtonGroup>
                  </Stack>
                </div>
                <div className="col-span-12">
                  {input === 'input' ? (
                    <TextField
                      id="queries"
                      label="Queries"
                      placeholder="Queries"
                      className="w-full"
                      value={queryString}
                      onChange={changeQueryString}
                      multiline
                      rows={12}
                      required
                    />
                  ) : (
                    <div className="rounded border">
                      {queries.map((query: string, index: number) => {
                        return (
                          <div
                            key={`query-${query}`}
                            className="grid grid-cols-12 border-b py-4"
                          >
                            <div className="col-span-1 text-center">
                              {index + 1}
                            </div>
                            <div className="col-span-11 truncate">
                              <p className="truncate">{query}</p>
                            </div>
                          </div>
                        );
                      })}
                    </div>
                  )}
                </div>
              </div>
            </form>
          </div>
        </section>
        <section>
          <div className="rounded border p-8 shadow-2xl">
            <div className="grid grid-cols-12 items-center gap-8">
              <div className="col-span-12">
                <div className="flex items-center justify-between">
                  <h2 className="text-lg uppercase">Results</h2>
                  <Button
                    type="button"
                    variant="outlined"
                    onClick={downloadCSV}
                  >
                    Download CSV
                  </Button>
                </div>
              </div>
              <div className="col-span-12">
                <TableContainer
                  component={Paper}
                  sx={{ maxHeight: 440 }}
                  className="border"
                >
                  <Table stickyHeader sx={{ minWidth: 650 }} aria-label="table">
                    <TableHead>
                      <TableRow>
                        <TableCell align="center">No.</TableCell>
                        <TableCell sx={{ maxWidth: '300px' }}>Query</TableCell>
                        <TableCell align="right">
                          Collectable Total Mentions
                        </TableCell>
                        <TableCell align="right">Total Mentions</TableCell>
                      </TableRow>
                    </TableHead>
                    <TableBody>
                      {loading < 100 ? (
                        <TableRow className="border-0 border-none">
                          <TableCell
                            align="center"
                            colSpan={4}
                            className="border-0 border-none uppercase"
                          >
                            {loading}%
                          </TableCell>
                        </TableRow>
                      ) : (
                        <>
                          {results.length > 0 ? (
                            results.map((result: Result, index: number) => {
                              const resultQuery = result.query || '';
                              return (
                                <TableRow
                                  key={`result-${resultQuery.replaceAll(
                                    ' ',
                                    ''
                                  )}`}
                                  sx={{
                                    '&:last-child td, &:last-child th': {
                                      border: 0
                                    }
                                  }}
                                >
                                  <TableCell align="center">
                                    {index + 1}
                                  </TableCell>
                                  <TableCell
                                    component="th"
                                    scope="row"
                                    sx={{ maxWidth: '300px' }}
                                  >
                                    <p className="truncate">{resultQuery}</p>
                                  </TableCell>
                                  <TableCell align="right">
                                    {result.total_collectable_mentions}
                                  </TableCell>
                                  <TableCell align="right">
                                    {result.total_mentions}
                                  </TableCell>
                                </TableRow>
                              );
                            })
                          ) : (
                            <TableRow className="border-0 border-none">
                              <TableCell
                                align="center"
                                colSpan={4}
                                className="border-0 border-none uppercase"
                              >
                                No Data
                              </TableCell>
                            </TableRow>
                          )}
                        </>
                      )}
                    </TableBody>
                  </Table>
                </TableContainer>
              </div>
            </div>
          </div>
        </section>
      </main>
    </LocalizationProvider>
  );
};

export default SocialHeatPage;
