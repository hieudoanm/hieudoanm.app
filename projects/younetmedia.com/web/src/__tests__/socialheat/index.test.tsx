import { render } from '@testing-library/react';
import { SocialHeatPage } from '../../pages/socialheat';

jest.mock('next/router', () => {
  return { useRouter: jest.fn() };
});

describe('SocialHeatPage', () => {
  it('renders correctly', () => {
    const { container } = render(<SocialHeatPage />);
    expect(container).toMatchSnapshot();
  });
});
