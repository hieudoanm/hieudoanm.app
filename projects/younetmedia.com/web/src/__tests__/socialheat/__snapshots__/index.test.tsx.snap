// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialHeatPage renders correctly 1`] = `
<div>
  <nav
    class="border-b shadow"
  >
    <div
      class="container mx-auto px-8 py-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <div>
          <h1
            class="text-lg uppercase"
          >
            Buzz Metrics
          </h1>
        </div>
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium uppercase css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Log Out
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </nav>
  <main
    class="container mx-auto p-8"
  >
    <section
      class="mb-8"
    >
      <div
        class="rounded border p-8 shadow-2xl"
      >
        <form>
          <div
            class="grid grid-cols-12 items-center gap-8"
          >
            <div
              class="col-span-12"
            >
              <div
                class="flex items-center justify-between"
              >
                <h2
                  class="text-lg uppercase"
                >
                  Queries
                </h2>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="submit"
                >
                  Run Queries
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="col-span-12"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="pin"
                  id="pin-label"
                >
                  PIN
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="pin"
                    placeholder="PIN"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        PIN
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="col-span-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root w-full css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="topicId"
                  id="topicId-label"
                >
                  Topic ID
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="topicId"
                    placeholder="Topic ID"
                    required=""
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        Topic ID
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="col-span-6"
            >
              <div
                class="MuiFormControl-root MuiTextField-root w-full css-z3c6am-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r2:"
                  id=":r2:-label"
                >
                  From Date
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    inputmode="text"
                    placeholder="DD/MM/YYYY"
                    type="text"
                    value="20/12/2023"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date, selected date is Dec 20, 2023"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        From Date
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="col-span-6"
            >
              <div
                class="MuiFormControl-root MuiTextField-root w-full css-z3c6am-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r4:"
                  id=":r4:-label"
                >
                  To Date
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r4:"
                    inputmode="text"
                    placeholder="DD/MM/YYYY"
                    type="text"
                    value="20/12/2023"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date, selected date is Dec 20, 2023"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        To Date
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="col-span-12"
            >
              <div
                class="MuiStack-root css-6rg9iv-MuiStack-root"
              >
                <div
                  class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
                  role="group"
                >
                  <button
                    aria-pressed="true"
                    class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeLarge MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-2tvszk-MuiButtonBase-root-MuiToggleButton-root"
                    tabindex="0"
                    type="button"
                    value="input"
                  >
                    Input
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-pressed="false"
                    class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeLarge MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-2tvszk-MuiButtonBase-root-MuiToggleButton-root"
                    tabindex="0"
                    type="button"
                    value="table"
                  >
                    Table
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="col-span-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root w-full css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="queries"
                  id="queries-label"
                >
                  Queries
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-dpjnhs-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    id="queries"
                    placeholder="Queries"
                    required=""
                    rows="12"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Queries
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </section>
    <section>
      <div
        class="rounded border p-8 shadow-2xl"
      >
        <div
          class="grid grid-cols-12 items-center gap-8"
        >
          <div
            class="col-span-12"
          >
            <div
              class="flex items-center justify-between"
            >
              <h2
                class="text-lg uppercase"
              >
                Results
              </h2>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Download CSV
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="col-span-12"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root border css-1uzl8r8-MuiPaper-root-MuiTableContainer-root"
            >
              <table
                aria-label="table"
                class="MuiTable-root MuiTable-stickyHeader css-1nfbgt6-MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignCenter MuiTableCell-sizeMedium css-dwuj3p-MuiTableCell-root"
                      scope="col"
                    >
                      No.
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-6vgeb-MuiTableCell-root"
                      scope="col"
                    >
                      Query
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-i02g0k-MuiTableCell-root"
                      scope="col"
                    >
                      Collectable Total Mentions
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-i02g0k-MuiTableCell-root"
                      scope="col"
                    >
                      Total Mentions
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root border-0 border-none css-1q1u3t4-MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium border-0 border-none uppercase css-1yhpg23-MuiTableCell-root"
                      colspan="4"
                    >
                      0
                      %
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </section>
  </main>
</div>
`;
